src/taipy/core/notification/notifier.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from queue import Queue
+from typing import List, Tuple
+
+from .event import Event
+from .registration import Registration
+from .topic import Topic
+
+
+class Notifier:
+    # TODO add a lock on _registration
+    _registrations: List[Registration] = []
+
+    @classmethod
+    def register(cls, entity_type, entity_id, operation, attribute_name, time_to_live) -> Tuple[str, Queue]:
+        register_id = "gui"  # TODO generate an id to return so the client can unregister.
+        registration = Registration(register_id, entity_type, entity_id, operation, attribute_name, time_to_live)
+        # TODO lock before updating cls._registrations
```

Comment:
I dont think it's necessary to lock here as the registrations should be done before starting a core event consumer thread to get the queue, we won't have any conflict of resources at this point.